src/platform_impl/ios/uikit/gesture_recognizer.rs

```diff
@@ -0,0 +1,141 @@
+use crate::platform_impl::platform::view::WinitView;
+use icrate::Foundation::{CGFloat, NSInteger, NSObject, NSUInteger};
+use objc2::encode::{Encode, Encoding};
+use objc2::rc::Id;
+use objc2::runtime::Sel;
+use objc2::{extern_class, extern_methods, msg_send_id, mutability, ClassType};
+
+// https://developer.apple.com/documentation/uikit/uigesturerecognizer
+extern_class!(
+    #[derive(Debug, PartialEq, Eq, Hash)]
+    pub(crate) struct UIGestureRecognizer;
+
+    unsafe impl ClassType for UIGestureRecognizer {
+        type Super = NSObject;
+        type Mutability = mutability::InteriorMutable;
+    }
+);
+
+extern_methods!(
+    unsafe impl UIGestureRecognizer {
+        #[method(state)]
+        pub fn state(&self) -> UIGestureRecognizerState;
+
+        #[method(initWithTarget:action:)]
+        pub fn initWithTarget(&self, target: WinitView, action: Sel);
```

Comment:
Unsafe because the selector must match the correct method.

```suggestion
        pub unsafe fn initWithTarget_action(&self, target: &AnyObject, action: Sel);
```

Same goes for similar methods on the subclasses.

Reply:
marked it as unsafe, but not sure what do you mean by that, it's the only one in all of the iOS impl